File: Examples/android/CMeSdkJava/app_rework/src/main/java/me/connect/sdk/java/sample/db/dao/ConnectionDao.java
package me.connect.sdk.java.sample.db.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import me.connect.sdk.java.sample.db.entity.Connection;

@Dao
public interface ConnectionDao {
    @Query("SELECT * FROM connection")
    List<Connection> getAll();

    @Query("SELECT * FROM connection WHERE id = :id")
    Connection getById(int id);

    @Insert
    void insertAll(Connection... connections);

    @Update
    void update(Connection connection);

    @Delete
    void delete(Connection connection);
}

File: Examples/android/CMeSdkJava/app_rework_kt/src/main/kotlin/me/connect/sdk/java/samplekt/db/dao/ConnectionDao.kt
package me.connect.sdk.java.samplekt.db.dao


import androidx.room.*

import me.connect.sdk.java.samplekt.db.entity.Connection


@Dao
interface ConnectionDao {
    @Query("SELECT * FROM connection")
    suspend fun getAll(): List<Connection>

    @Query("SELECT * FROM connection WHERE id = :id")
    suspend fun getById(id: Int): Connection

    @Insert
    suspend fun insertAll(vararg connections: Connection)

    @Update
    suspend fun update(connection: Connection)

    @Delete
    suspend fun delete(connection: Connection)
}
